recipes/cyclonedds/all/conanfile.py

```diff
@@ -0,0 +1,174 @@
+                                                " gcc 6 or higher.")
+            if version < "9" and compiler.get_safe("libcxx") == "libstdc++":
+                raise ConanInvalidConfiguration("gcc < 9 with libstdc++ not supported")
+            if version == "6":
+                self.output.warn("Cyclone DDS package is compiled with gcc 6, it is"\
+                                 " recommended to use 7 or higher")
+                self.output.warn("GCC 6 will build with warnings.")
+        elif compiler == "clang":
+            if compiler.get_safe("libcxx") == "libstdc++":
+                raise ConanInvalidConfiguration("clang with libstdc++ not supported")
+            if version == "7.0" and compiler.get_safe("libcxx") == "libc++" and \
+               self.options.shared and self.settings.build_type == "Debug":
+                raise ConanInvalidConfiguration("shared Debug with clang 7.0 and"\
+                                                " libc++ not supported")
+
+    def source(self):
+        files.get(self,**self.conan_data["sources"][self.version], strip_root=True,
+                 destination=self._source_subfolder)
+
+    def layout(self):
+        cmake_layout(self)
+
+    def generate(self):
+
+        tc = CMakeToolchain(self)
+        # ToDo : determine how to do in conan : 
+        # - idlc is a code generator that is used as tool (and so not cross compiled)
+        # - other tools like ddsperf is cross compiled for target
+        # - maybe separate package like cyclonedds_idlc 
+        tc.variables["BUILD_IDLC"]            = False
+        tc.variables["BUILD_IDLC_TESTING"]    = False
+        tc.variables["BUILD_DDSPERF"]         = False
+        tc.variables["BUILD_IDLC_TESTING"]    = False
+        # variables which effects build
+        tc.variables["BUILD_SHARED_LIBS"]     = self.options.shared
+        tc.variables["ENABLE_SSL"]            = self.options.ssl 
+        tc.variables["ENABLE_SHM"]            = self.options.shm
+        tc.variables["ENABLE_SECURITY"]       = self.options.security
+        tc.generate()
+
+        cd = CMakeDeps(self)
+        cd.generate()
+
+    def build(self):
+        files.apply_conandata_patches(self)
+        cmake = CMake(self)
+        cmake.configure(build_script_folder=self._source_subfolder)
+        cmake.build()
+
+    def package(self):
+        cmake = CMake(self)
+        cmake.install()
+        self.copy("LICENSE", src=self._source_subfolder, dst="licenses")
+        files.rmdir(self, os.path.join(self.package_folder, "share"))
+        files.rmdir(self, os.path.join(self.package_folder, "lib","pkgconfig"))
+        files.rmdir(self, os.path.join(self.package_folder, "lib","cmake","CycloneDDS"))
+
+    def package_info(self):
+        self._create_cmake_module_alias_targets(
+                os.path.join(self.package_folder, self._module_file_rel_path),
+                { "CycloneDDS::ddsc" : "cyclonedds::ddsc"})
```

Comment:
```suggestion
        self.cpp_info.filenames["cmake_find_package"] = "cyclonedds"
        self.cpp_info.filenames["cmake_find_package_multi"] = "cyclonedds"
        self.cpp_info.names["cmake_find_package"] = "CycloneDDS"
        self.cpp_info.names["cmake_find_package_multi"] = "CycloneDDS"
        self.cpp_info.components["CycloneDDS"].names["cmake_find_package"] = "ddsc"
        self.cpp_info.components["CycloneDDS"].names["cmake_find_package_multi"] = "ddsc"
        self.cpp_info.components["CycloneDDS"].libs = ["foobar"]
```

It will generate `cyclonedds-config.cmake` with `CycloneDDS::ddsc` as target